packages/atomic/src/components/commerce/product-template-components/atomic-product-children/atomic-product-children.tsx

```diff
@@ -0,0 +1,124 @@
+import {Schema, StringValue} from '@coveo/bueno';
+import {Product, ChildProduct} from '@coveo/headless/commerce';
+import {
+  Component,
+  h,
+  Element,
+  Prop,
+  Event,
+  EventEmitter,
+  State,
+} from '@stencil/core';
+import {
+  InitializableComponent,
+  InitializeBindings,
+} from '../../../../utils/initialization-utils';
+import {CommerceBindings} from '../../atomic-commerce-interface/atomic-commerce-interface';
+import {ProductContext} from '../product-template-decorators';
+
+export interface SelectChildProductEventArgs {
+  childPermanentId: string;
+  parentPermanentId: string;
+}
+
+/**
+ * @internal
+ * The `atomic-product-children` component renders a section that allows the user to select a nested product (e.g., a color variant of a given product).
+ *
+ * This component leverages the [product grouping](https://docs.coveo.com/en/l78i2152/) feature.
+ */
+@Component({
+  tag: 'atomic-product-children',
+  styleUrl: 'atomic-product-children.pcss',
+  shadow: false,
+})
+export class AtomicProductChildren
+  implements InitializableComponent<CommerceBindings>
+{
+  @InitializeBindings() public bindings!: CommerceBindings;
+  @ProductContext() private product!: Product;
+
+  @Element() hostElement!: HTMLElement;
+
+  @State() private children: ChildProduct[] = [];
+  @State() private activeChildId: string = '';
+
+  public error!: Error;
+
+  // TODO: We could leverage the totalNumberOfRelatedProducts property on the Product to show the number of available
+  // children (e.g., ...and 3 more). This would require some UX design though.
+
+  /**
+   * The non-localized label to display for the product children section.
+   */
+  @Prop() public label: string = 'available-in';
```

Comment:
We could omit that prop, and then the user would be responsible for adding the label themselves with an `atomic-commerce-text` component.

However, including the label is consistent with the UX mockup.

Reply:
I made it so that passing an empty string as a label won't render the label part at all, so it's possible for users to get rid of it this way.